ext/opcache/jit/zend_jit_ir.c

```diff
@@ -10294,22 +10298,6 @@ static int zend_jit_do_fcall(zend_jit_ctx *jit, const zend_op *opline, const zen
 		ir_GUARD_NOT(ir_LOAD_A(jit_EG_exception(jit)),
 			jit_STUB_ADDR(jit, jit_stub_icall_throw));
 
-		// TODO: Can we avoid checking for interrupts after each call ???
-		if (trace && jit->last_valid_opline != opline) {
-			int32_t exit_point = zend_jit_trace_get_exit_point(opline + 1, ZEND_JIT_EXIT_TO_VM);
-
-			exit_addr = zend_jit_trace_get_exit_addr(exit_point);
-			if (!exit_addr) {
-				return 0;
-			}
-		} else {
-			exit_addr = NULL;
-		}
-
-		if (!zend_jit_check_timeout(jit, opline + 1, exit_addr)) {
-			return 0;
-		}
-
```

Comment:
What does the surrounding code look like? I might have expected this to use `jne` to jump to some other place which then jumps back to `.L1`, since this is cold (and I believe we marked it as such in the JIT IR). This way, the CPU impact is isolated to when the branch is taken (not often). Something like:

```asm
   cmp EG(vm_interrupts), $0
   jne .L2
.L1
   ; ...
   
.L2
   movabs %rax, $zend_interrupt_or_timeout)
   call %rax
   jmp .L1
```

For the happy paths without interrupts, this would be the same performance (minus possible cache effects from sightly increased code size which I think is unlikely) as master. But maybe there's a lot of asm in `; ...` and it puts the cold code pretty far away. I would personally still be okay with penalizing the interrupt path, but I can see why the optimizer might not make that choice.

Reply:
In any case, even with the proposed optimisation this makes the code worse (at least bigger).

I don't like to do this just for a new feature of third-party project.
This will affect every PHP installation (without your profiler).

Reply:
We could use an exit when `zend_interrupt_function` is null (so the generated code is not changed), and a call to `zend_interrupt_or_timeout` otherwise. This way there is no regression when `zend_interrupt_function` is null.

Reply:
It's a good idea. I think many users will have the `pcntl` extension, though, which sets `pcntl_interrupt_function`. Are we okay with special casing `pcntl_interrupt_function`? The code would need some reworking but we logically the check is `pcntl_loaded && zend_interrupt_function == pcntl_interrupt_function  && orig_interrupt_function == NULL` then it's the same as `!zend_interrupt_function`.

Reply:
@dstogov Can I get your opinion on the current patch which implements basic `zend_interrupt_function` specialization in JIT?